fl4health/parameter_exchange/sparse_coo_parameter_exchanger.py

```diff
@@ -0,0 +1,154 @@
+import math
+from logging import INFO
+from typing import Callable, Dict, List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, NDArrays
+from torch import Tensor
+from torch.nn.modules import Module
+
+from fl4health.parameter_exchange.parameter_packer import SparseCooParameterPacker
+from fl4health.parameter_exchange.partial_parameter_exchanger import PartialParameterExchanger
+
+
+class SparseCooParameterExchanger(PartialParameterExchanger[Tuple[NDArrays, NDArrays, List[str]]]):
+    def __init__(self, sparsity_level: float, score_gen_function: Callable[..., Dict[str, Tensor]]) -> None:
+        """
+        Parameter exchanger for sparse tensors.
+
+        This exchanger is responsible for selecting an arbitrary subset of a model's parameters
+        via some selection criterion and then packaging them into the COO sparse tensor format for exchanging.
+
+        Args:
+            sparsity_level (float): The level of sparsity. Must be between 0 and 1.
+            score_gen_function (Callable[..., Dict[str, Tensor]]): Function that is responsible for
+            generating a score for every parameter inside a model in order to facilitate parameter selection.
+
+            In most cases, this function takes as inputs a current model and an initial model,
+            and it returns a dictionary that maps the name of each of the current model's tensors to
+            another tensor which contains the parameter scores.
+        """
+        assert 0 < sparsity_level <= 1
+        self.set_sparsity_level(sparsity_level)
+        self.parameter_packer: SparseCooParameterPacker = SparseCooParameterPacker()
+        self.set_score_gen_function(score_gen_function)
+
+    def set_sparsity_level(self, sparsity_level: float) -> None:
+        """This method exists to allow for setting new sparsity levels throughout training."""
+        self.sparsity_level = sparsity_level
+
+    def set_score_gen_function(self, score_gen_function: Callable[..., Dict[str, Tensor]]) -> None:
```

Comment:
I think this setter function can be removed and the `score_gen_function` property can be set in the `__init__`